src/vsr/replica.zig

```diff
@@ -1594,6 +1594,8 @@ pub fn ReplicaType(
             assert(message.header.replica < self.replica_count);
             assert(message.header.operation != .reserved);
 
+            self.replicate(message);
```

Comment:
This is probably too early: we shouldn't replicate messages which are not the tip of the log. So at least we should move this beyond `is_repair`. And perhaps "older view" case also shouldn't replicate? Though I don't have a clear mental model of that case, so would be good to double-check when that happens.

Reply:
Ah, indeed. This should at least be moved beyond `on_repair` given we'd needlessly be replicating the prepares that we requested from another replica, which is just plain wrong!

Regarding all of the other checks, my understanding was that replication entails simply _forwarding_ the prepare to the next replica in the chain; whether that prepare should and can be processed at a particular backup (and appended to the WAL) is governed (and protected) by all the checks that we have in place in on_prepare. For example, even if we receive a prepare from the future, we can forward it to the next replica in the chain instead of dropping it entirely, because the next replica _may_ be in a view that can accept the prepare (and we are going to drop it locally anyway, thanks to our checks in on_prepare). Thoughts?